### PR TITLE
plan: fix bug when outer join's condition is false

### PR DESCRIPTION
**Pull request by hanfei1991**
When the query is "select sum(-1) from t a left join t b on false", we rewirte it as "select sum(agg0) from t a join (select sum(-1) as agg0 from t where false) b". The table t will never return any record, but sum(-1) without group by will return NULL, which break our default values "-1". To avoid this bug, we should rewrite it to "select sum(agg0) from t a join (select sum(-1) as agg0 from t where false group by 1) b"
@shenli @coocood @zimulala @XuHuaiyu @tiancaiamao PTAL

### CODE REVIEW

File: plan/aggregation_push_down.go [hunk 1]
```diff
@@ -224,6 +225,11 @@ func (a *aggPushDownSolver) tryToPushDownAgg(aggFuncs []expression.AggregationFu
 	}
 	agg.AggFuncs = newAggFuncs
 	agg.SetSchema(schema)
+	if len(agg.GroupByItems) == 0 {
```
Comment (XuHuaiyu) [inline plan/aggregation_push_down.go]: comment for this.

**Comment by shenli** [comment]
LGTM

File: plan/aggregation_push_down.go [hunk 2]
```diff
@@ -225,6 +225,8 @@ func (a *aggPushDownSolver) tryToPushDownAgg(aggFuncs []expression.AggregationFu
 	}
 	agg.AggFuncs = newAggFuncs
 	agg.SetSchema(schema)
+	// If agg has no group by items, it will return a default values, which may cause some bug.
```
Comment (XuHuaiyu) [inline plan/aggregation_push_down.go]: no group by items --> no group-by item
a default values ?
some bug --> some bugs

**Comment by hanfei1991** [comment]
@XuHuaiyu PTAL

**Comment by XuHuaiyu** [comment]
LGTM

**Pull request merged**